src/lines-yardstick.coffee

```diff
@@ -32,32 +33,27 @@ class LinesYardstick
     lineOffset = lineNode.getBoundingClientRect().left
     targetLeft += lineOffset
 
-    textNodeStartColumn = 0
-    for textNode in textNodes
-      {length: textNodeLength, textContent: textNodeContent} = textNode
-      textNodeRight = @clientRectForRange(textNode, 0, textNodeLength).right
+    textNodeComparator = (textNode, position) =>
+      {length: textNodeLength} = textNode
+      rangeRect = @clientRectForRange(textNode, 0, textNodeLength)
+      return -1 if rangeRect.right < position
+      return 1 if rangeRect.left > position
+      return 0
 
-      if textNodeRight > targetLeft
-        characterIndex = 0
-        while characterIndex < textNodeLength
-          if isPairedCharacter(textNodeContent, characterIndex)
-            nextCharacterIndex = characterIndex + 2
-          else
-            nextCharacterIndex = characterIndex + 1
+    textNodeIndex = binarySearch(textNodes, targetLeft, textNodeComparator)
 
-          rangeRect = @clientRectForRange(textNode, characterIndex, nextCharacterIndex)
+    if textNodeIndex >= 0
+      textNodeStartColumn = textNodes
+        .slice(0, textNodeIndex)
```

Comment:
Can we avoid duplicating the array contents in this case? We could scan through all the characters and use an imperative approach, which would make this code path even faster.